code/go/internal/validator/spec.go

```diff
@@ -138,12 +139,26 @@ func (s Spec) rules(rootSpec spectypes.ItemSpec) validationRules {
 		if rule.until != nil && !s.version.LessThan(rule.until) {
 			continue
 		}
+
+		if rule.types != nil && !contains(rule.types, pkgType) {
+			continue
+		}
+
 		validationRules = append(validationRules, rule.fn)
 	}
 
 	return validationRules
 }
 
+func contains(elems []string, v string) bool {
```

Comment:
Nit. Call this like the type it checks.
```suggestion
func stringSliceContains(elems []string, v string) bool {
```
Or use generics :slightly_smiling_face: 
```suggestion
func sliceContains[T any](elems []T, v T) bool {
```

Or maybe we can use this library that has chances of ending up in the stdlib https://pkg.go.dev/golang.org/x/exp/slices#Contains

Reply:
Trying the generics option, there is an error that I don't see why it is a "set":
> internal/validator/spec.go:155:6: invalid operation: a == v (incomparable types in type set)

I will go for the keep using strings (first option).

Reply:
Ah, it seems it shouldn't be `any`, but `comparable`, sorry.
```
func sliceContains[T comparable](elems []T, v T) bool {
```